timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/v5/graphql/datafetchers/SummaryPropDataRetriever.java

```diff
@@ -31,31 +31,33 @@ public SummaryPropDataRetriever(String userConfiguredPredicate, List<SummaryProp
     ).findFirst();
 
     if (userConfigured.isPresent()) {
-      return Optional.ofNullable(createTypedValue(userConfigured.get(), dataSet));
+      return Optional.of(createTypedValue(userConfigured.get(), dataSet));
     } else { // fallback to default summary props
-      Optional<CursorQuad> quad = Optional.empty();
       for (SummaryProp prop : defaultProperties) {
-        for (String path : prop.getPath()) {
-          String uri = quad.isPresent() ? quad.get().getObject() : source.getSubjectUri();
-          quad = quadStore.getQuads(uri, path, Direction.OUT, "").findAny();
-
-          if (!quad.isPresent()) {
-            break;
-          }
-        }
+        Optional<CursorQuad> quad = getQuad(prop.getPath(), source.getSubjectUri(), quadStore);
         if (quad.isPresent()) {
-          break;
+          return Optional.of(createTypedValue(quad.get(), dataSet));
         }
       }
-
-      Optional<CursorQuad> foundData = quad;
-      if (foundData.isPresent()) {
-        return Optional.ofNullable(createTypedValue(foundData.get(), dataSet));
-      }
     }
     return Optional.empty();
   }
 
+  private Optional<CursorQuad> getQuad(List<String> path, String uri, QuadStore quadStore) {
+    /*
+     * A collect of the data might look a bit nicer, but that will cause all the data to be loaded from the database.
+     * This way only the data needed is retrieved.
+     * See https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html#StreamOps for more info.
+     */
+    return quadStore.getQuads(uri, path.get(0), Direction.OUT, "").map(quad -> {
```

Comment:
You should manually check if the triple's object is a value (i.e. the quads valuetype.isPresent()) and only recurse of it's not. If it is, then you should return Optional.empty if the path's size > 1

Otherwise `<subj> <pred> "http://someValue"^^xsd:anyUri` will then jump to `<http://someValue> <otherPred> ...`  whereas that branch should return no value.